nautobot/core/ui/breadcrumbs.py

```diff
@@ -0,0 +1,500 @@
+        prepend_items: BreadcrumbItemsType = None,
+        append_items: BreadcrumbItemsType = None,
+        template: str = "inc/breadcrumbs.html",
+    ):
+        """
+        Initialize the Breadcrumbs configuration.
+
+        Args:
+            items (Optional[dict[str, list[BreadcrumbItem]]]): Default breadcrumb items for each action.
+            prepend_items (Optional[dict[str, list[BreadcrumbItem]]]): Items to prepend to action's breadcrumbs.
+            append_items (Optional[dict[str, list[BreadcrumbItem]]]): Items to append to action's breadcrumbs.
+            template (str): The template used to render the breadcrumbs.
+        """
+        self.template = template
+        self.items: BreadcrumbItemsType = copy.deepcopy(DEFAULT_BREADCRUMBS)
+        if items:
+            self.items.update(items)
+        # TODO: after breadcrumbs implementation verify if we need append/prepend feature
+        self.prepend_items: BreadcrumbItemsType = prepend_items or {}
+        self.append_items: BreadcrumbItemsType = append_items or {}
+
+    def get_breadcrumbs_items(self, context: Context):
+        """
+        Compute the list of breadcrumb items for the given context.
+
+        Items are determined based on the `view_action` in context. This includes:
+        - Prepend items
+        - Default items
+        - Append items
+
+        Args:
+            context (Context): The view or template context that holds `view_action` and related state.
+
+        Returns:
+            (list[tuple[str, str]]): A list of (url, label) tuples representing breadcrumb entries.
+        """
+        action = context.get("view_action", "list")
+        detail = context.get("detail", False)
+        items = self.get_items_for_action(self.items, action, detail)
+        prepend_items = self.get_items_for_action(self.prepend_items, action, detail)
+        append_items = self.get_items_for_action(self.append_items, action, detail)
+        all_items = prepend_items + items + append_items
+        return [item.as_pair(context) for item in all_items if item.should_render(context)]
+
+    def filter_breadcrumbs_items(self, items: list[tuple[str, str]], context: Context) -> list[tuple[str, str]]:
+        """
+        Filters out all items that both label and url are None or empty str.
+
+        Args:
+            items (list[tuple[str, str]]): breadcrumb items pair.s
+            context (Context): The view or template context.
+
+        Returns:
+            (list[tuple[str, str]]): A list of filtered breadcrumb items pairs.
+        """
+        return [(url, label) for url, label in items if self.is_not_both_blank(url, label)]
+
+    @staticmethod
+    def is_not_both_blank(url: str, label: str) -> bool:
+        """
+        Check if both values are not empty / None.
+        We're still assuming that user intentionally would like to put some breadcrumbs
+        with just label or empty string / only whitespace with url.
```

Comment:
I can see the value of a `label` without a `url`, but the idea of a `url` without a `label` seems wrong to me.